src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -314,4 +316,36 @@ public void validateFinancialPlans_missingInputs_throwsParseException() {
         assertThrows(ParseException.class, () -> ParserUtil.validateFinancialPlans(
                 Arrays.asList("", VALID_FINANCIAL_PLAN_1)));
     }
+
+    @Test
+    void isValidDay_validDate_shouldReturnTrue() {
+        // November has 30 days
+        YearMonth yearMonth = YearMonth.of(2022, 11);
+        String date = "15-11-2022";
+        assertTrue(ParserUtil.isValidDay(yearMonth, date));
+    }
+
+    @Test
+    void isValidDay_invalidDate_shouldReturnFalse() {
+        // February has 28 days
+        YearMonth yearMonth = YearMonth.of(2022, 2);
+        String date = "31-02-2022";
+        assertFalse(ParserUtil.isValidDay(yearMonth, date));
+    }
+
+    @Test
+    void parseValidAppointmentDate_validDate_shouldParseSuccessfully() throws ParseException {
+        String validDateString = "15-11-2022 14:30";
+        try {
+            ParserUtil.parseValidAppointmentDate(validDateString);
+        } catch (ParseException e) {
+            fail();
+        }
+    }
+
+    @Test
+    void parseValidAppointmentDate_invalidDate_shouldThrowParseException() {
+        String invalidDateString = "31-02-2022 14:30";
+        assertThrows(ParseException.class, () -> ParserUtil.parseValidAppointmentDate(invalidDateString));
+    }
```

Comment:
Can consider inputs such as 29th Feb for leap year, 29th Feb for non leap year, 31st for month like April instead of Febuary to make testing more comprehensive